fastlane_core/lib/fastlane_core/itunes_transporter.rb

```diff
@@ -569,14 +690,33 @@ def upload(app_id = nil, dir = nil, package_path: nil, asset_path: nil)
       password_placeholder = @jwt.nil? ? 'YourPassword' : nil
       jwt_placeholder = @jwt.nil? ? nil : 'YourJWT'
 
-      command = @transporter_executor.build_upload_command(@user, @password, actual_dir, @provider_short_name, @jwt)
-      UI.verbose(@transporter_executor.build_upload_command(@user, password_placeholder, actual_dir, @provider_short_name, jwt_placeholder))
+      # Handle AppStore Connect API
+      use_api_key = !@api_key.nil?
+      api_key_placeholder = nil unless use_api_key
+      api_key_placeholder = { key_id: "YourKeyID", issuer_id: "YourIssuerID" } if use_api_key
+
+      if use_api_key
+        # Specified p8 needs to be generated to call altool
+        tmp_p8_file_path = File.join(Dir.tmpdir, "AuthKey_#{@api_key[:key_id]}.p8")
```

Comment:
I'd suggest to instead use `Dir.mktmpdir` to create an intermediate and guaranteed-unique temporary folder and generate the key file inside it. Or use `TempFile` to create the file.

(PS: In either case you can provide a prefix (like `deliver-`) to use for the autogenerate unique filename which could help identify it on disk… even though it'll get deleted soon after so it doesn't really matter)

If you go that route you'll have to declare `tmp_p8_file_path = nil` before the `begin`, then set it to a value on this line, so that you'll be able to use the same `tmp_p8_file_path` value in your `ensure` block to delete it.

Reply:
Done! https://github.com/fastlane/fastlane/pull/20631/commits/c82e99a896e5a07978c8d114948ca3ee6c88ef3b